pkg/drivers/kic/oci/oci.go

```diff
@@ -640,28 +640,35 @@ func iptablesFileExists(ociBin string, nameOrID string) bool {
 	return true
 }
 
-// TODO: comment
-func DockerHost(driver string) string {
+// RuntimeHost returns the ip/hostname where OCI daemon service for driver is running
+// For Podman it's always DefaultBindIPV4
+// For Docker return the host part of DOCKER_HOST environment variable if set
+// or DefaultBindIPV4 otherwise
+func RuntimeHost(driver string) string {
 	if driver != Docker {
 		return DefaultBindIPV4
 	}
-	if v := DockerMachineHost(driver); v != "" {
-		return v
+	if dh := os.Getenv(constants.DockerHostEnv); dh != "" {
+		if u, err := url.Parse(dh); err == nil {
+			if u.Host != "" {
+				return u.Hostname()
+			}
+		}
 	}
 	return DefaultBindIPV4
 }
 
-// TODO: comment
-func DockerMachineHost(driver string) string {
+// IsExternalRuntimeHost returns whether or not the OCI runtime is running on an external/virtual host
+// For Podman driver it's always false for now
```

Comment:
the rules for podman should be the same as for docker, when it comes to external vs. local
only the default transport differs, if it is tcp: or ssh: (as opposed to unix: for the local socket)

eventually it is going to end up looking _very_ similar, once we get rid of the legacy protocols
then it will always be the same unix socket, possibly tunneled over a ssh connection first...

Reply:
Agreed. I just want to reduce the scope of this PR and implement support of external docker endpoints here, and later do the same for Podman.

Reply:
I am okay with adding that in a following up PR